docs/BUILD.bazel

```diff
@@ -2,49 +2,72 @@ load("@bazel_skylib//rules:write_file.bzl", "write_file")
 load("@bazel_skylib//rules:diff_test.bzl", "diff_test")
 load("@io_bazel_stardoc//stardoc:stardoc.bzl", "stardoc")
 
+# Dictionary of key-value pairs where:
+# key := {bzl_library target}
+# value := ({output MD file}, {rule template}, {footer file})
 _DOC_SRCS = {
-    "//extras:embed_data": "extras.md",
+    "//extras:embed_data": ("extras.md", "rule_subsection.vm", "footers/empty.md"),
+    "//go:core_docs": ("core.md", "rule_subsection.vm", "footers/core_docs.md"),
 }
 
+# Generate MD from .bzl
 [
     stardoc(
-        name = out.replace(".md", "_docgen"),
-        out = out,
+        name = out[0].replace(".md", "-docgen"),
+        out = out[0].replace(".md", "-docgen.md"),
         input = input + ".bzl",
         deps = [input],
+        rule_template = out[1]
     )
     for [
         input,
         out,
     ] in _DOC_SRCS.items()
 ]
 
+# Append footers onto generated MD
 [
-    diff_test(
-        name = "check_" + out,
-        failure_message = "Please run \"bazel run //docs:update\"",
-        # Source file
-        file1 = "//go:" + out,
-        # Output from stardoc rule above
-        file2 = out,
+    genrule(
+        name = out[0].replace(".md", "-linked"),
+        srcs = [out[0].replace(".md", "-docgen.md"), out[2]],
```

Comment:
this is ugly and hard to understand, just use conventions so that the loop variable `out` is a string not a list

Reply:
good point